### ISSUE TITLE
[iOS][MauiEmbedding] Uno.Extensions.Maui.MauiEmbeddingException

### ISSUE THREAD

**Issue by agneszitte** [unoplatform/uno.extensions#2240]
## Current behavior
Create an app using: `dotnet new unoapp -preset blank -maui -o MauiEmbeddingApp` with latest template [5.3.0-dev.19](https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=120036&view=results).

The application is building fine but we are getting this exception when launching the app for iOS:
   `$exception.Message.Message: Uno.Extensions.Maui.MauiEmbeddingException: 'CoreWindow is missing _window internal field required to access UIWindow, you may need an updated version of the Uno.Extension.Maui.WinUI package.'`
   
   (And it is already using the latest dev version *Uno.Extension.Maui.WinUI 4.2.0-dev.13*)
  <details>

<summary>Exception details to expand:</summary>

  ```
   $exception.InnerException: null
   ```
   ```
   $exception.StackTrace.StackTrace:
      at Uno.Extensions.Maui.MauiEmbedding.<>c.<RegisterPlatformServices>b__0_0(IServiceProvider sp)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitFactory(FactoryCallSite factoryCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext, Microsoft.Extensions.DependencyInjection, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].VisitCallSiteMain(ServiceCallSite callSite, RuntimeResolverContext argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext, Microsoft.Extensions.DependencyInjection, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[System.Object, System.Private.CoreLib, Version=8.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].VisitCallSite(ServiceCallSite callSite, RuntimeResolverContext argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(ServiceIdentifier serviceIdentifier)
   at System.Collections.Concurrent.ConcurrentDictionary`2[[Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceIdentifier, Microsoft.Extensions.DependencyInjection, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60],[Microsoft.Extensions.DependencyInjection.ServiceProvider.ServiceAccessor, Microsoft.Extensions.DependencyInjection, Version=8.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].GetOrAdd(ServiceIdentifier key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(ServiceIdentifier serviceIdentifier, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType)
   at Microsoft.Maui.MauiContext.WrappedServiceProvider.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[UIWindow](IServiceProvider provider)
   at Uno.Extensions.Maui.MauiEmbedding.InitializeApplicationMainPage(IApplication iApp)
   at Uno.Extensions.Maui.MauiEmbedding.InitializeMauiEmbeddingApp(MauiApp mauiApp, Application app)
   at Uno.Extensions.Maui.MauiEmbedding.BuildMauiApp(MauiAppBuilder builder, Application app, Window window)
   at Uno.Extensions.Maui.MauiEmbedding.<>c__DisplayClass3_0`1[[TestCustomApp530dev19.MauiControls.App, TestCustomApp530dev19.MauiControls, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<UseMauiEmbedding>b__0()
   at Uno.Extensions.Maui.UnoServiceProviderFactory.CreateBuilder(IServiceCollection services)
   at Uno.Extensions.Hosting.Internal.ServiceFactoryAdapter`1[[System.IServiceProvider, System.ComponentModel, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].CreateBuilder(IServiceCollection services)
   at Uno.Extensions.Hosting.HostBuilder.CreateServiceProvider()
   at Uno.Extensions.Hosting.HostBuilder.Build()
   at Uno.Extensions.Hosting.ApplicationBuilder.Build()
   at Uno.Extensions.ApplicationBuilderExtensions.<>c__DisplayClass0_0`1[[TestCustomApp530dev19.Presentation.Shell, TestCustomApp530dev19, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].<NavigateAsync>b__1()
   at Uno.Extensions.ServiceProviderExtensions.BuildAndInitializeHostAsync(Window window, FrameworkElement viewHost, Func`1 buildHost, String initialRoute, Type initialView, Type initialViewModel, Func`3 initialNavigate)
   at Uno.Extensions.ApplicationBuilderExtensions.<NavigateAsync>d__0`1[[TestCustomApp530dev19.Presentation.Shell, TestCustomApp530dev19, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext()
   at TestCustomApp530dev19.App.OnLaunched(LaunchActivatedEventArgs args) in C:\Dev\Samples\Cli\TestCustomApp530dev19\TestCustomApp530dev19\App.xaml.cs:line 100
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at Foundation.NSAsyncSynchronizationContextDispatcher.Apply() in /Users/builder/azdo/_work/1/s/xamarin-macios/src/Foundation/NSAction.cs:line 176
   at UIKit.UIApplication.UIApplicationMain(Int32 argc, String[] argv, IntPtr principalClassName, IntPtr delegateClassName) in /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:line 58
   at UIKit.UIApplication.Main(String[] args, Type principalClass, Type delegateClass) in /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:line 94
   at TestCustomApp530dev19.iOS.EntryPoint.Main(String[] args) in C:\Dev\Samples\Cli\TestCustomApp530dev19\TestCustomApp530dev19\Platforms\iOS\Main.iOS.cs:line 12
   ```
	```
	   Call Stack:
	    0x11 in System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw	C#
	 	0x6 in System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0	C#
	 	0xC in Foundation.NSAsyncSynchronizationContextDispatcher.Apply at /Users/builder/azdo/_work/1/s/xamarin-macios/src/Foundation/NSAction.cs:176,5	C#
	 	0x38 in UIKit.UIApplication.xamarin_UIApplicationMain	C#
	 	0xE in UIKit.UIApplication.UIApplicationMain at /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:58,4	C#
	 	0x51 in UIKit.UIApplication.Main at /Users/builder/azdo/_work/1/s/xamarin-macios/src/UIKit/UIApplication.cs:94,4	C#
	>	0xD in TestCustomApp530dev19.iOS.EntryPoint.Main at C:\Dev\Samples\Cli\TestCustomApp530dev19\TestCustomApp530dev19\Platforms\iOS\Main.iOS.cs:12,9	C#
	```

</details> 

## Expected behavior

No exception

## How to reproduce it (as minimally and precisely as possible)

- Create an app using: `dotnet new unoapp -preset blank -maui -o MauiEmbeddingApp` with latest template [5.3.0-dev.19](https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=120036&view=results)
- Try to launch the application for iOS
- Notice the exception

Sample app used for the tests: [MauiEmbeddingAppiOSTest.zip](https://github.com/unoplatform/uno.extensions/files/14936316/MauiEmbeddingAppiOSTest.zip)

## Environment

Versions used (by default when creating the app):
```
    "Uno.Sdk": "5.3.0-dev.111"
    <UnoExtensionsVersion>4.2.0-dev.13</UnoExtensionsVersion>
    <UnoToolkitVersion>5.2.0-dev.56</UnoToolkitVersion>
    <UnoThemesVersion>4.2.0-dev.30</UnoThemesVersion>
    <UnoCSharpMarkupVersion>5.3.0-dev.11</UnoCSharpMarkupVersion>
    Microsoft.WindowsAppSDK: 1.5.240311000
    Microsoft.Windows.SDK.BuildTools: 10.0.22621.3233
    MauiVersion: 8.0.6
```
--
Even with latest versions having the same exception for iOS:
```
    "Uno.Sdk": "5.3.0-dev.168"
    <UnoExtensionsVersion>4.2.0-dev.13</UnoExtensionsVersion>
    <UnoToolkitVersion>6.1.0-dev.7</UnoToolkitVersion>
    <UnoThemesVersion>5.1.0-dev.8</UnoThemesVersion>
    <UnoCSharpMarkupVersion>5.3.0-dev.11</UnoCSharpMarkupVersion>
    Microsoft.WindowsAppSDK: 1.5.240311000
    Microsoft.Windows.SDK.BuildTools: 10.0.22621.3233
    MauiVersion: 8.0.6
```
(Related sample app with latest versions: [MauiEmbeddingBlankApp.zip](https://github.com/unoplatform/uno.extensions/files/14936928/MauiEmbeddingBlankApp.zip))

- [X] iOS
- [ ] macOS (AppKit)
- [ ] Mac Catalyst
- [ ] Android
- [ ] WebAssembly
- [ ] Windows
- [ ] Skia (WPF)
- [ ] Skia (GTK on Linux/macOS/Windows)
- [ ] Skia (Linux Framebuffer)
- [ ] Build tasks

Visual Studio:

- [ ] 2019 (version: )
- [X] 2022 (version: 17.9.2)
- [ ] Visual Studio Code (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )

**Comment by jeromelaban** [comment]
This looks like a internal change that was introduced by expecting a specific field to exist. We should be having a proper API to provide this information. /cc @MartinZikmund

**Comment by agneszitte** [comment]
This issue is fixed with latest dev and latest stable 5.2 packages
